File: frontend/public/examples/javascript/file_io.js
// Writing to a file in JavaScript (Node.js)

const fs = require('fs');

// Write to a file
fs.writeFileSync('output.txt', 'Hello, World!\n');
fs.appendFileSync('output.txt', 'This is a test file.\n');

// Append to a file
fs.appendFileSync('output.txt', 'Appended line.\n');

// Read from a file
const content = fs.readFileSync('output.txt', 'utf-8');
console.log(content);

// Async version
async function fileOperations() {
    const fsPromises = require('fs').promises;

    await fsPromises.writeFile('async_output.txt', 'Async Hello!\n');
    const data = await fsPromises.readFile('async_output.txt', 'utf-8');
    console.log(data);
}

fileOperations();

File: examples/cpp/file_io.cpp
// Writing to a file in C++

#include <iostream>
#include <fstream>
#include <string>

int main() {
    // Write to a file
    std::ofstream outFile("output.txt");
    if (outFile.is_open()) {
        outFile << "Hello, World!" << std::endl;
        outFile << "This is a test file." << std::endl;
        outFile.close();
    }

    // Append to a file
    std::ofstream appendFile("output.txt", std::ios::app);
    if (appendFile.is_open()) {
        appendFile << "Appended line." << std::endl;
        appendFile.close();
    }

    // Read from a file
    std::ifstream inFile("output.txt");
    if (inFile.is_open()) {
        std::string line;
        while (std::getline(inFile, line)) {
            std::cout << line << std::endl;
        }
        inFile.close();
    }

    return 0;
}
